mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/DefaultUserQuotaRootResolver.java

```diff
@@ -57,6 +57,7 @@ public QuotaRoot forUser(User user) {
     @Override
     public QuotaRoot getQuotaRoot(MailboxPath mailboxPath) {
         Preconditions.checkArgument(!mailboxPath.getNamespace().contains(SEPARATOR), "Namespace should not contain " + SEPARATOR);
+        Preconditions.checkArgument(mailboxPath.getUser() != null, "Username should not be null");
         Preconditions.checkArgument(!mailboxPath.getUser().contains(SEPARATOR), "Username should not contain " + SEPARATOR);
         User user = User.fromUsername(mailboxPath.getUser());
```

Comment:
@chibenwa I think we should reconsider this task. If we accepting `MailboxPath#getUser()` as `null` we'll receive `NullPointerException` here.

Reply:
Why?

Can't we succeed to generate this even without user, and apply this normalization transformation only when we get a user?

Reply:
Do you mean like this:

```
return QuotaRoot.quotaRoot(mailboxPath.getNamespace(), Optional.empty());
```

Reply:
Yes, that is what I propose.

Reply:
Yes, indeed looks logical. Will update it soon and thanks for your guidance.

Reply:
You are very welcome ;-)